manifests/config/passenger/fragment.pp

```diff
@@ -7,23 +7,25 @@
 #  $ssl_content:: content of the ssl virtual host fragment
 #
 define foreman::config::passenger::fragment(
-  $content='',
-  $ssl_content='',
+  $content=undef,
+  $ssl_content=undef,
 ) {
   require foreman::config::passenger
 
-  file { "${apache::confd_dir}/05-foreman.d/${name}.conf":
-    ensure  => present,
-    content => $content,
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0644',
-    notify  => Class['apache::service'],
+  if $content {
```

Comment:
I think I'd prefer an else, otherwise we'd be specifying a file resource with mode, owner and content, but with ensure => absent, which feels strange.  (And Puppet would be justified in throwing a compilation or runtime error I'd say.)

I'll add a check for `''` too, good point.